npk/src/archive.rs

```diff
@@ -190,83 +195,66 @@ impl TryFrom<SerdeHashes> for Hashes {
     }
 }
 
-const MANIFEST: &str = "manifest.yaml";
-const SIGNATURE: &str = "signature.yaml";
-const FS_IMAGE: &str = "fs.img";
-
-pub struct ArchiveReader<'a> {
+pub struct ArchiveReader {
     archive: zip::ZipArchive<std::io::BufReader<std::fs::File>>,
-    signing_keys: &'a HashMap<String, ed25519_dalek::PublicKey>,
-}
-
-pub fn read_manifest(
-    npk: &Path,
-    signing_keys: &HashMap<String, ed25519_dalek::PublicKey>,
-) -> Result<Manifest, Error> {
-    let mut archive_reader = ArchiveReader::new(&npk, &signing_keys)?;
-    archive_reader.extract_manifest_from_archive()
 }
 
-impl<'a> ArchiveReader<'a> {
-    pub fn new(
-        npk: &Path,
-        signing_keys: &'a HashMap<String, ed25519_dalek::PublicKey>,
-    ) -> Result<Self, Error> {
+impl ArchiveReader {
+    pub fn new(npk: &Path) -> Result<Self, Error> {
         let file =
             std::fs::File::open(&npk).map_err(|_e| Error::CouldNotOpenFile(PathBuf::from(npk)))?;
-
         let reader: std::io::BufReader<std::fs::File> = std::io::BufReader::new(file);
         let archive: zip::ZipArchive<std::io::BufReader<std::fs::File>> =
             zip::ZipArchive::new(reader).map_err(Error::Zip)?;
-        Ok(Self {
-            archive,
-            signing_keys,
-        })
+        Ok(Self { archive })
+    }
+
+    pub fn extract_manifest_version(&mut self) -> Result<Version, Error> {
+        let zip_comment = String::from_utf8(self.archive.comment().to_vec())
+            .map_err(|e| Error::ArchiveError(format!("Failed to read manifest version {}", e)))?;
+        let split_comment: Vec<_> = zip_comment.split(' ').collect();
```

Comment:
This could be done with this pattern:
```rust
let mut split = zip_comment.split(' ');
while let Some(key) = split.next() {
  if let Some(value) = split.next() {
   // todo
 } else {
   return Err("Missing value for key);
 }
}
```